src/github.com/couchbase/sync_gateway/rest/config.go

```diff
@@ -494,7 +494,7 @@ func ParseCommandLine() {
 			config.LogFilePath = logFilePath
 		}
 
-		if skipRunModeValidation != nil {
+		if *skipRunModeValidation == true {
```

Comment:
This change looks like it's just applying the command line parameter to the config, when a config is used.

It was failing for me when I was trying to run with no command line parameter, and only the config setting.  Is that combination working for you?